File: spec/java/src/io/kaitai/struct/spec/TestNestedSameName.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.NestedSameName;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

public class TestNestedSameName extends CommonSpec {
    @Test
    public void testNestedSameName() throws Exception {
        NestedSameName r = NestedSameName.fromFile(SRC_DIR + "repeat_n_struct.bin");

        assertEquals(r.mainData().mainSize(), 2);
        assertEquals(r.mainData().foo().data(), new byte[] { 0x10, 0, 0, 0 });
    }
}

File: spec/cpp_stl/test_nested_same_name.cpp
#include <boost/test/unit_test.hpp>

#include <nested_same_name.h>

#include <iostream>
#include <fstream>
#include <vector>

BOOST_AUTO_TEST_CASE(test_nested_same_name) {
    std::ifstream ifs("src/repeat_n_struct.bin", std::ifstream::binary);
    kaitai::kstream ks(&ifs);
    nested_same_name_t* r = new nested_same_name_t(&ks);

    BOOST_CHECK_EQUAL(r->main_data()->main_size(), 2);
    BOOST_CHECK_EQUAL(r->main_data()->foo()->data(), std::string("\x10\0\0\0", 4));

    delete r;
}
